### PR TITLE
fix(core): decode observation predictions with the same safe scale as encode

### PR DESCRIPTION
**Pull request by mtcow**
## Summary
`targets()` normalizes observation targets by `safe_scale = max(observation_scale, 1e-6)`, but `_prediction_and_raw_diagnostics()` denormalized predictions by the raw `observation_scale`. Below `1e-6` a perfectly fitted learner reconstructed `observation_scale / max(observation_scale, 1e-6)` of its trained delta (e.g. `1e-9` scale reconstructed `1e-3` of the target). The reward leg was already paired on the raw config value with no floor; only the observation leg was unpaired.

## Fix
Decode by the same `safe_scale` used at encode:

```python
safe_scale = jnp.maximum(obs_scale, jnp.asarray(1e-6, dtype=jnp.float32))
decoded_next_observation = jnp.where(
    self._config.predict_delta,
    safe_obs + normalized_delta * safe_scale,
    normalized_delta * safe_scale,
)
```

## Verification
Numpy round-trip check (JAX not installed in the sandbox):

```text
scale=0.001: 固定后还原误差=0.00e+00 (旧版=0.00e+00)
scale=1e-07: 固定后还原误差=0.00e+00 (旧版=9.00e-01)
scale=1e-09: 固定后还原误差=0.00e+00 (旧版=9.99e-01)
```

### CODE REVIEW

**Review by agent-cortex — CHANGES_REQUESTED** [review]
## Review — fix(core): decode observation predictions with the same safe scale as encode

I verified the headline change statically and reproduced its numerical claim independently. The core fix is correct, but I'm requesting changes on two concrete, defensible grounds: the PR's disclosed scope does not match its diff, and the one change the PR is titled for is the only change shipped without a regression test.

### What I verified (headline fix is correct)
`alberta_framework/core/world_model.py`:
- Encode in `targets()` (L809/812-813) normalizes by `safe_scale = max(observation_scale, 1e-6)`.
- The reward leg is already symmetric (encode L816 / decode L860 both use `reward_scale`).
- Before this PR, `_prediction_and_raw_diagnostics()` decoded the observation leg with the **raw** `obs_scale` (old L848-849), so for any config with `observation_scale` below `1e-6` a perfectly fitted learner reconstructed only `observation_scale / max(observation_scale, 1e-6)` of its trained delta. The change decodes with the same `safe_scale`, restoring the round-trip.

Reachability confirmed: `observation_scale` is validated only with `positive=True` (L368 → `_validated_config_float`), so positive sub-`1e-6` scales are valid configs; the defect is reachable, not hypothetical.

Independent reproduction of the encode/decode round-trip logic (pure Python; see "execution note" below for why the real module could not run):

```
   obs_scale true_delta |      OLD err      NEW err
       1e-03   5.00e-04 |   0.0000e+00   0.0000e+00   (OLD rel err=0.000)
       1e-06   5.00e-07 |   0.0000e+00   0.0000e+00   (OLD rel err=0.000)
       1e-07   5.00e-08 |   4.5000e-08   0.0000e+00   (OLD rel err=0.900)
       1e-09   5.00e-10 |   4.9950e-10   0.0000e+00   (OLD rel err=0.999)
       1e-12   5.00e-13 |   5.0000e-13   0.0000e+00   (OLD rel err=1.000)
```

This matches the PR description's claimed errors (0.9 @ 1e-7, 0.999 @ 1e-9) and confirms the fix is a no-op on the common `>= 1e-6` path — a clean, well-scoped 2-line correction.

### Blocking concern 1 — undisclosed scope; PR is not reviewable as one unit
The title and the entire PR description describe **only** the `world_model.py` observation-decode fix, but the diff (`c9aba7b...ee395fc`, 798/-50 across 22 files) actually contains **10 independent bugfixes**, each closing a different issue:

```
ec7b3b3 fix(core): align STOMP importance ratios with Gumbel behavior policy   (Closes #2136)
8a49ba3 fix(benchmarks): validate rule-discovery shard arm and stage identity  (Closes #2134)
0c47d92 fix(steps): bound Step 7 planning and Step 9 dreaming loop lengths      (Closes #2214)
b848f84 fix(core): floor_and_renormalize_probabilities returns a simplex ...    (Closes #2238)
8647e12 fix(utils): significance markers follow each result's stored alpha      (Closes #2227)
57438cf fix(core): cap working-memory decay rate cardinality ...                (Closes #2220)
f7c7858 fix(core): bound stacked-horde demon count ...                          (Closes #2225)
e4e3982 fix(core): make UPGD gradient alignment scale-free
bb9eef9 fix(eval): rescale flad noise projection so tiny gradients ...
ee395fc fix(core): decode observation predictions with the same safe scale ...  (this PR's title)
```

Consequences:
- Merging this PR silently closes **seven issues** (#2136, #2134, #2214, #2238, #2227, #2220, #2225) that the PR body never mentions, and merges live-path behavioral changes a reviewer reading the title/description would not know are present — e.g. `options.py::_epsilon_greedy_action_probabilities` now returns `softmax(q/1e-6)` instead of the isclose-tie split (changes STOMP importance ratios), and `behavior_model.py::floor_and_renormalize_probabilities` gains a uniform fallback. These are plausible fixes, but each is an independent change to a distinct hot path that warrants its own focused review, and each already carries its own issue and its own test file.
- Bundling defeats bisect/revert and contradicts the repo's CONTRIBUTING guidance (contributions "judged by the uncertainty they resolve ... not by diff size").

Requested repair: scope PR #2408 to the `world_model.py` change so it matches its title, and submit the other nine fixes as their own PRs (they are already cleanly separated by commit and issue). Each is then individually reviewable and safely closeable.

### Blocking concern 2 — the headline fix ships with no regression test
Every other change in this PR adds a dedicated test (test_options, test_behavior_model, test_export, test_stacked_horde_cardinality_bounds, test_working_memory_cardinality_bounds, test_step7/9_hostile_validation, test_upgd_scale_free_alignment, test_rule_discovery_summary_hostile). The one change the PR is named for — the observation decode — adds **none**. The "Verification" in the description is a NumPy re-implementation pasted into the PR body (partly in mixed-language output), not a test in the suite, so nothing guards against regression.

Requested repair: add a round-trip test to `tests/test_action_conditioned_world_model.py` with `observation_scale` below `1e-6` (e.g. `1e-9`) asserting that a fitted-delta prediction decodes back to the target — this both locks in the fix and would have caught the original bug.

### Execution note
Deep execution proof against the real module is blocked in the sandbox: it runs Python 3.11 with no `pip`/`numpy`/`jax`, and the repo requires `requires-python >= 3.12` plus the JAX stack. I therefore reproduced the encode/decode arithmetic directly (above) rather than importing `ActionConditionedWorldModel`; the focused JAX tests (`pytest tests/test_action_conditioned_world_model.py`) could not be executed here and should be confirmed in CI.

### Summary
The core two-line fix is correct, reachable, and reproduced. Please (1) split this into per-issue PRs so the scope matches the title and the seven `Closes #` auto-closes are each reviewed, and (2) add the missing world_model round-trip regression test. Once scoped and tested, the observation-decode change looks approvable on its own.

---
AI provider/model: anthropic / claude-opus-4-8
Client / agent tooling: pi-coding-agent
Contribution skill revision: elizaOS/eliza@1f236fd1f58f9a5be09e5ac10c439040a7ea6a3c:packages/skills/skills/contribute-to-eliza
Attribution status: self-reported
— [eliza-swarm]

**Comment by lalalune** [comment]
Closing after full diff review: This stacked PR changes the decode scale to max(scale,1e-6), silently overriding the configured observation scale, while inheriting unrelated caps, export rules, and unsafe reciprocal gradient normalization. Focused encode/decode fixes #2793/#2794 preserve the declared scale and are under review. The inherited components are being resolved through their narrow PRs.
